autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding.py

```diff
@@ -0,0 +1,108 @@
+from typing import Dict, Optional, Tuple, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+import ConfigSpace.hyperparameters as CSH
+from ConfigSpace import EqualsCondition
+
+import numpy as np
+import pandas as pd
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class BagOfWordEncoder(AutoSklearnPreprocessingAlgorithm):
+    def __init__(
+        self,
+        ngram_range: Optional[int] = None,
+        min_df_choice: Optional[str] = None,
+        min_df_absolute: Optional[int] = None,
+        min_df_relative: Optional[float] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.ngram_range = ngram_range
+        self.random_state = random_state
+        self.min_df_choice = min_df_choice
+        self.min_df_absolute = min_df_absolute
+        self.min_df_relative = min_df_relative
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'BagOfWordEncoder':
+
+        if isinstance(X, pd.DataFrame):
+            # define a CountVectorizer for every feature (implicitly defined by order of columns,
+            # maybe change the list
+            # to a dictionary with features as keys)
+            if self.min_df_choice == "min_df_absolute":
+                self.preprocessor = CountVectorizer(min_df=self.min_df_absolute,
+                                                    ngram_range=(1, self.ngram_range))
+            elif self.min_df_choice == "min_df_relative":
+                self.preprocessor = CountVectorizer(min_df=self.min_df_relative,
+                                                    ngram_range=(1, self.ngram_range))
+            else:
+                raise KeyError()
+            for feature in X.columns:
+                self.preprocessor = self.preprocessor.fit(X[feature])
```

Comment:
This seems like a major issue, essentially the final `preprocessor` is only fit on the last text column. For data with multiple "string" columns, this would make things quite poor?

Reply:
Seems we need either one preprocessor per text column, or if using all columns in one preprocessor, then need something like `self.preprocessor.fit(X)`. If that doesn't work then 
```python
all_text = itertools.chain.from_iterable(X[col] for col in X.columns)
self.preprocessor.fit(all_text)
```

Reply:
I thought i tested it that it just updates the vocabulary_ but i turns out that this is wrong. Then we have to create a transformer for every column. The question is to stick with this version or using the "minimal version" which only uses disticnt transformers for each column

Reply:
fit operation is not in-place, therefore the preprocessor is unfitted after the command above. I would recommand:
`self.preprocessor = self.preprocessor.fit(all_text)`